### ISSUE TITLE
Data list view: sticky elements hide focused elements and tooltips appear on top of sticky elements

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#56939]
### Description

I just noticed introduced two important usability and accessibility problems that need to be fixed. Cc @jameskoster @jasmussen 

1. When navigating the table content with the keyboard, the currently focused element may be entirely hidden behind the sticky header / footer. This is a violation of the WCAG 2.2 [Focus not obscured](https://www.w3.org/TR/WCAG22/#focus-not-obscured-minimum) success criterium and needs to be fixed.
2. The button tooltips show on toop of the sticky header / footer when scrolling the table.

## Focus obscured
[Focus not obscured](https://www.w3.org/TR/WCAG22/#focus-not-obscured-minimum) requires the currently focused element to be _not entirely hidden_. After https://github.com/WordPress/gutenberg/pull/56157 when tabbing through the focusable elements in the data views table, the currently focused element may be entirely hidden.
In the screenshots below, I'm illustrating the case when they are _partially_ hidden for more clarity. However, depending on the scroll position they may be _entirely_ hidden. This needs to be ficed.

In the WCAG 2.2 criterium [Understanding note](https://www.w3.org/WAI/WCAG22/Understanding/focus-not-obscured-minimum.html#examples), there is a specific example about sticky footers. Suggested remediation: use [scroll padding](https://www.w3.org/TR/css-scroll-snap/#propdef-scroll-padding).

Screenshot: the shown tooltip actually belongs to a button in the above row that is entirely hidden behind the sticky header:

![Screenshot 2023-12-11 at 10 12 49](https://github.com/WordPress/gutenberg/assets/1682452/741b1cbd-1ca4-4f31-becc-e7789aaf47b3)

Screenshot: focused link partially hidden behind the sticky footer (depending on scroll position it may be entirely hidden):

![Screenshot 2023-12-11 at 10 14 20](https://github.com/WordPress/gutenberg/assets/1682452/d18ae8b2-dc62-453d-94eb-1e22cba539c0)

Screenshot: same for buttons:

![Screenshot 2023-12-11 at 10 16 52](https://github.com/WordPress/gutenberg/assets/1682452/1fb2c8da-cb9f-4664-b809-9ad20653410a)

## Tooltips on top of sticky header and footer
Whjen a tooltip appears and users scroll the table, the tooltip is shown on top of the sticky header and footer, triggering visual glitches that should be remediated. See animated GIF below:
![scrolling tooltips](https://github.com/WordPress/gutenberg/assets/1682452/e30aaee4-94da-4896-84ae-8c492a02108a)

### Step-by-step reproduction instructions

- In the Site editor navigate to Pages > Manage all pages
- Use the keyboard to navigate through focusable elements within the table.
- Observe some focused elements may be entirely hidden.
- Observe visual glitches ot the tooltips appearing on top of sticky header and footer

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by jameskoster** [comment]
I opened https://github.com/WordPress/gutenberg/pull/56946 to address the focus issue.

The tooltip one might be tricky to handle. cc @ciampo for any ideas.

**Comment by ciampo** [comment]
> The tooltip one might be tricky to handle. cc @ciampo for any ideas.

It is a bit tricky. Trying out an approach in #57026

**Comment by youknowriad** [comment]
Closing this issue as the main issue scrolling is solved, the tooltip is trickier due to popovers behavior but less important anyway.
